File: app/src/main/java/com/example/test/di/DatabaseModule.java
package com.example.test.di;

import android.content.Context;

import com.example.test.data.AppDatabase;
import com.example.test.data.users.UserDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.android.qualifiers.ApplicationContext;
import dagger.hilt.components.SingletonComponent;

@Module
@InstallIn(SingletonComponent.class)
public class DatabaseModule {

    @Singleton
    @Provides
    public static AppDatabase provideAppDatabase(@ApplicationContext Context context) {
        return AppDatabase.getDatabase(context);
    }

    @Provides
    public static UserDao providePlantDao(AppDatabase appDatabase) {
        return appDatabase.getUsersDao();
    }
}

File: app/src/main/java/com/example/test/di/DatabaseModule.kt
package com.example.test.data.di

import android.content.Context
import com.example.test.data.AppDatabase
import com.example.test.data.users.UserDao
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@InstallIn(SingletonComponent::class)
@Module
class DatabaseModule {

    @Singleton
    @Provides
    fun provideAppDatabase(@ApplicationContext context: Context): AppDatabase {
        return AppDatabase.getDatabase(context)
    }

    @Provides
    fun providePlantDao(appDatabase: AppDatabase): UserDao {
        return appDatabase.getUsersDao()
    }
}
